### PR TITLE
Command service sh-m update

### PR DESCRIPTION
**Pull request by kamilsa**
### Description of the Change

Introduces shared model to the command service.

### Benefits

Validation is done when transport object is converted into shared model object

### Possible Drawbacks 

Redundant validation is happening when transaction goes to transaction processor.

### CODE REVIEW

File: irohad/torii/impl/command_service.cpp [hunk 1]
```diff
@@ -17,7 +17,11 @@
 
 #include <thread>
 
+#include <validators/default_validator.hpp>
+#include <validators/transaction_validator.hpp>
```
Comment (l4l) [inline irohad/torii/impl/command_service.cpp]: In `<`, `>`

File: irohad/torii/impl/command_service.cpp [hunk 2]
```diff
@@ -39,44 +43,67 @@ namespace torii {
...
         [this](
             std::shared_ptr<iroha::model::TransactionResponse> iroha_response) {
           // Find response in cache
-          auto res = cache_->findItem(iroha_response->tx_hash);
+          auto tx_hash = shared_model::crypto::Hash(iroha_response->tx_hash);
+          auto res = cache_->findItem(tx_hash);
           if (not res) {
             iroha::protocol::ToriiResponse response;
-            response.set_tx_hash(iroha_response->tx_hash);
+            response.set_tx_hash(tx_hash.toString());
```
Comment (l4l) [inline irohad/torii/impl/command_service.cpp]: Use `crypto::toBinaryString` from blob.hpp, `toString` returns human-readable

File: irohad/torii/impl/command_service.cpp [hunk 3]
```diff
@@ -39,44 +43,67 @@ namespace torii {
...
-               iroha::cache::Cache<std::string,
-                                   iroha::protocol::ToriiResponse>>()) {
+        cache_(std::make_shared<CacheType>()) {
     // Notifier for all clients
     tx_processor_->transactionNotifier().subscribe(
         [this](
             std::shared_ptr<iroha::model::TransactionResponse> iroha_response) {
           // Find response in cache
-          auto res = cache_->findItem(iroha_response->tx_hash);
+          auto tx_hash = shared_model::crypto::Hash(iroha_response->tx_hash);
```
Comment (l4l) [inline irohad/torii/impl/command_service.cpp]: Are you sure that `iroha_response->tx_hash` is in binary form?

File: irohad/torii/impl/command_service.cpp [hunk 4]
```diff
@@ -39,44 +43,67 @@ namespace torii {
...
-    auto iroha_tx = pb_factory_->deserialize(tx);
-    auto tx_hash = iroha::hash<iroha::protocol::Transaction>(tx).to_string();
+  void CommandService::Torii(const iroha::protocol::Transaction &request) {
+    shared_model::proto::TransportBuilder<
+        shared_model::proto::Transaction,
+        shared_model::validation::DefaultTransactionValidator>()
+        .build(request)
+        .match(
+            [this](
+                // success case
```
Comment (l4l) [inline irohad/torii/impl/command_service.cpp]: Either remove or move this comment somewhere else

File: irohad/torii/impl/command_service.cpp [hunk 5]
```diff
@@ -39,44 +43,67 @@ namespace torii {
...
+                const iroha::expected::Value<shared_model::proto::Transaction>
+                    &iroha_tx) {
+              auto tx_hash = iroha_tx.value.hash();
+              if (cache_->findItem(tx_hash)) {
+                return;
+              }
+              iroha::protocol::ToriiResponse response;
+              response.set_tx_hash(tx_hash.toString());
```
Comment (l4l) [inline irohad/torii/impl/command_service.cpp]: Again `crypto::toBinaryString`

File: irohad/torii/impl/command_service.cpp [hunk 6]
```diff
@@ -39,44 +43,67 @@ namespace torii {
...
+              cache_->addItem(tx_hash, response);
+              // Send transaction to iroha
+              tx_processor_->transactionHandle(
+                  std::shared_ptr<iroha::model::Transaction>(
+                      iroha_tx.value.makeOldModel()));
+            },
+            [this, &request](const auto &error) {
+              // stateless validation in transport validator failed
+
+              // TODO remove creation of invalid transaction
```
Comment (l4l) [inline irohad/torii/impl/command_service.cpp]: jira id, date, author?

File: irohad/torii/impl/command_service.cpp [hunk 7]
```diff
@@ -39,44 +43,67 @@ namespace torii {
...
+              auto tx_hash =
+                  shared_model::proto::Transaction(std::move(request)).hash();
```
Comment (l4l) [inline irohad/torii/impl/command_service.cpp]: UB is here, you move const ref

Comment (l4l) [inline irohad/torii/impl/command_service.cpp]: Why not just get hash from `protocol::Transaction`?

File: irohad/torii/impl/command_service.cpp [hunk 8]
```diff
@@ -39,44 +43,67 @@ namespace torii {
...
 
-    iroha::protocol::ToriiResponse response;
-    response.set_tx_hash(tx_hash);
-    response.set_tx_status(iroha::protocol::TxStatus::IN_PROGRESS);
+              iroha::protocol::ToriiResponse response;
+              response.set_tx_hash(tx_hash.toString());
```
Comment (l4l) [inline irohad/torii/impl/command_service.cpp]: `toBinaryString`

File: test/integration/client_test.cpp [hunk 9]
```diff
@@ -128,29 +130,17 @@ TEST_F(ClientServerTest, SendTxWhenValid) {
...
-  ASSERT_TRUE(model_tx.has_value());
-  auto status = client.sendTx(model_tx.value());
+  auto shm_tx = shared_model::proto::TransactionBuilder()
+      .creatorAccountId("some@account")
+      .txCounter(1)
+      .createdTime(iroha::time::now())
+      .setAccountQuorum("some@account", 2)
+      .build()
+      .signAndAddSignature(
+          shared_model::crypto::CryptoProviderEd25519Sha3::
```
Comment (l4l) [inline test/integration/client_test.cpp]: `DefaultCryptoAlgorithmType`

File: test/integration/client_test.cpp [hunk 10]
```diff
@@ -128,29 +130,17 @@ TEST_F(ClientServerTest, SendTxWhenValid) {
...
+          generateKeypair());
+
+  auto status = client.sendTx(*shm_tx.makeOldModel());
```
Comment (l4l) [inline test/integration/client_test.cpp]: Leak again

File: test/integration/client_test.cpp [hunk 11]
```diff
@@ -180,26 +170,16 @@ TEST_F(ClientServerTest, SendTxWhenInvalidJson) {
...
+  auto shm_tx = shared_model::proto::TransactionBuilder()
+      .creatorAccountId("some@account")
+      .txCounter(1)
+      .createdTime(iroha::time::now())
+      .setAccountQuorum("some@account", 2)
+      .build()
+      .signAndAddSignature(
+          shared_model::crypto::CryptoProviderEd25519Sha3::
+          generateKeypair());
+  auto tx = *shm_tx.makeOldModel();
```
Comment (l4l) [inline test/integration/client_test.cpp]: Leak agian

File: test/module/irohad/torii/torii_service_test.cpp [hunk 12]
```diff
@@ -149,6 +149,7 @@ class ToriiServiceTest : public testing::Test {
  */
 TEST_F(ToriiServiceTest, CommandClient) {
   iroha::protocol::TxStatusRequest tx_request;
+  tx_request.set_tx_hash("asdads");
```
Comment (l4l) [inline test/module/irohad/torii/torii_service_test.cpp]: Set valid hash size: `std::string(32, '\x0')`

File: test/module/irohad/torii/torii_service_test.cpp [hunk 13]
```diff
@@ -230,10 +231,16 @@ TEST_F(ToriiServiceTest, StatusWhenBlocking) {
...
-    payload->set_tx_counter(i);
-    payload->set_creator_account_id("accountA");
+    auto shm_tx = shared_model::proto::TransactionBuilder()
+                   .creatorAccountId("some@account")
+                   .txCounter(i + 1)
+                   .createdTime(iroha::time::now())
+                   .setAccountQuorum("some@account", 2)
+                   .build()
+                   .signAndAddSignature(
+                       shared_model::crypto::CryptoProviderEd25519Sha3::
```
Comment (l4l) [inline test/module/irohad/torii/torii_service_test.cpp]: `DefaultCryptoAlgorithmType`

File: test/module/irohad/torii/torii_service_test.cpp [hunk 14]
```diff
@@ -230,10 +231,16 @@ TEST_F(ToriiServiceTest, StatusWhenBlocking) {
...
   for (size_t i = 0; i < TimesToriiBlocking; ++i) {
-    auto new_tx = iroha::protocol::Transaction();
-    auto payload = new_tx.mutable_payload();
-    payload->set_tx_counter(i);
-    payload->set_creator_account_id("accountA");
+    auto shm_tx = shared_model::proto::TransactionBuilder()
+                   .creatorAccountId("some@account")
+                   .txCounter(i + 1)
+                   .createdTime(iroha::time::now())
+                   .setAccountQuorum("some@account", 2)
```
Comment (l4l) [inline test/module/irohad/torii/torii_service_test.cpp]: Extract account name const

File: irohad/torii/impl/command_service.cpp (hunk 1 above)
Comment (grimadas) [reply irohad/torii/impl/command_service.cpp]: Should be in `"``"`

Comment (l4l) [reply irohad/torii/impl/command_service.cpp]: Ehm, why? It's not system/third-party libs

Comment (l4l) [reply irohad/torii/impl/command_service.cpp]: oops, thanks

File: irohad/torii/impl/command_service.cpp (hunk 3 above)
Comment (kamilsa) [reply irohad/torii/impl/command_service.cpp]: yes

File: irohad/torii/impl/command_service.cpp [hunk 15]
```diff
@@ -27,6 +25,9 @@
 #include "model/converters/pb_common.hpp"
 #include "model/sha3_hash.hpp"
 #include "torii/command_service.hpp"
+#include "validators/default_validator.hpp"
+#include "validators/transaction_validator.hpp"
+#include "interfaces/base/hashable.hpp"
```
Comment (l4l) [inline irohad/torii/impl/command_service.cpp]: Misordered

File: test/integration/client_test.cpp [hunk 16]
```diff
@@ -131,17 +132,19 @@ TEST_F(ClientServerTest, SendTxWhenValid) {
...
+                    .creatorAccountId("some@account")
+                    .txCounter(1)
+                    .createdTime(iroha::time::now())
+                    .setAccountQuorum("some@account", 2)
+                    .build()
+                    .signAndAddSignature(
+                        shared_model::crypto::DefaultCryptoAlgorithmType::
+                            generateKeypair());
+
+  auto old_model = shm_tx.makeOldModel();
```
Comment (l4l) [inline test/integration/client_test.cpp]: better use unique_ptr

File: test/integration/client_test.cpp [hunk 17]
```diff
@@ -168,25 +171,24 @@ TEST_F(ClientServerTest, SendTxWhenInvalidJson) {
...
-
-  ASSERT_EQ(iroha_cli::CliClient(Ip, Port).sendTx(tx).answer,
+  // creating stateless invalid tx
+  auto shm_tx = TestTransactionBuilder()
+                    .creatorAccountId("some@account")
+                    .txCounter(1)
+                    .createdTime(iroha::time::now())
+                    .setAccountQuorum("some@@account", 2)
+                    .build();
+  auto *old_tx = shm_tx.makeOldModel();
```
Comment (l4l) [inline test/integration/client_test.cpp]: same

File: test/integration/client_test.cpp [hunk 18]
```diff
@@ -168,25 +171,24 @@ TEST_F(ClientServerTest, SendTxWhenInvalidJson) {
...
+
+  ASSERT_EQ(iroha_cli::CliClient(Ip, Port).sendTx(*old_tx).answer,
             iroha_cli::CliClient::OK);
-  ASSERT_EQ(iroha_cli::CliClient(Ip, Port)
-                .getTxStatus(iroha::hash(tx).to_string())
-                .answer.tx_status(),
-            iroha::protocol::TxStatus::STATELESS_VALIDATION_FAILED);
+  auto tx_hash = shm_tx.hash();
+  ASSERT_EQ(
+      iroha_cli::CliClient(Ip, Port)
```
Comment (l4l) [inline test/integration/client_test.cpp]: Is it really need to open connection each time?

Comment (kamilsa) [reply test/integration/client_test.cpp]: That is how Torii works at the moment. Fixing that out of scope of the task(

**Pull request merged**